seeders/20240303194232-roles.js

```diff
@@ -5,12 +5,14 @@ const allModels = require('../models/index'); // getting all models
 
 /** @type {import('sequelize-cli').Migration} */
 module.exports = {
-  async up (queryInterface, Sequelize) {
+  async up(queryInterface, Sequelize) {
     try {
       for (const role of rolesData) {
+
         const existingRole = await allModels.roles.findOne({ where: { name: role.name } });
         if (!existingRole) {
-          await allModels.roles.create({name: role.name});
+          const permissions = await allModels.permissions.create({ ...role.permissions })
```

Comment:
So here, you are creating permissions. Seeding permissions. So what if permissions are already created ?

Reply:
Permission data is given against a role. We're checking that if the role does not exist already then create the role and its permissions else neither role will be created nor permissions.